### ISSUE TITLE
before_update hook always runs

### ISSUE THREAD

**Issue by waderobson** [cloudreach/sceptre#632]
Hi I just ran across some unexpected behavior today.

After I've deploy a stack(create) the `before_update` hook runs even if there are no updates to perform.. 
The `after_update` hook however, only runs if there is updates performed. 

Is this intended behavior? 

```
hooks:
  before_update:
    - !cmd "echo heelllooooo"
  after_update:
    - !cmd "echo heelllooooo"
```

```
[2019-02-11 15:39:06] - marjory/base - Launching Stack
[2019-02-11 15:39:06] - marjory/base - Stack is in the UPDATE_COMPLETE state
heelllooooo
[2019-02-11 15:39:06] - marjory/base - Updating Stack
[2019-02-11 15:39:09] - marjory/base - No updates to perform.
```

**Comment by zaro0508** [comment]
The docs are not real clear but I believe the intended behavior is that hooks should only execute when there are stack updates.  I have confirmed that all hooks behave this way except for the `before_update` hook which seems to run even if there there are no stack updates.  I believe this is inconsistent behavior and should be a bug.

**Comment by waderobson** [comment]
That's my take too. 
[The examples](https://github.com/cloudreach/sceptre/blob/1cb1d003d872aa5039d3444b6c65e5f186531620/docs/docs/hooks.md#examples) suggest that you can run the provided asg hook before, to `suspend` and  and after,  to `resume`. This would create a problem if you're suspending all the time but only resuming if the stack had an update. 

I'll try and dig into this more and see if this is a regresion or not. I've only observed this with versions 2.0.0 and 2.0.1

**Comment by cornerman** [comment]
I agree with the proposed solution here, that `before_update` should only run if there are updates and that it should always have a succeeding `after_update`.

But can we have another `hook` that would be run every time a command stack is run? I am currently using `after_create` together with `before_update` in order to be always run, whenever the respective stack is launched. I need this to sync files into an s3-bucket which is created in this stack and I always want the current files to be there after running `launch`. Is there a better way?

**Comment by ngfgrant** [comment]
The intended behaviour is that the hooks are triggered before/after the command is initiated. I think the confusion is maybe related to #678 which will be fixed in 2.1.4 with better error handling which allows program to execute fully.
